protocol/minimum-base-fee.md

```diff
@@ -0,0 +1,99 @@
+# Minimum Base Fee
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | @niran                                             |
+| Created at         | 2025-07-01                                         |
+
+---
+
+# Purpose
+ 
+This document proposes establishing a configurable minimum base fee for OP Stack chains to shorten the length of priority fee auctions.
+
+# Summary
+
+We introduce a new configuration value that can be modified via `SystemConfig`:
+
+| Name             | Type    | Default | Meaning |
+|------------------|---------|---------|---------|
+| `minBaseFeeLog2` | `uint8` | `0`     | The log2 of the minimum base fee that can be charged for a transaction |
+
+These values are updated via a new function in `SystemConfig`:
+
+```solidity
+function setMinBaseFeeLog2(uint8 minBaseFeeLog2) external onlyOwner;
```

Comment:
It may be worth adding some info around the trust/security properties of allowing the chain operator to configure this. In particular do we need a maximum value (u8 is already fairly limiting but can still set a pretty high base fee because its log2). I _think_ it probably works out o not changing trust assumptions because a malicious chain operator could set a very large L1 fee scalar and all these fees are circumvented via deposit transactions so it can't introduce a censorship risk. Seems like we should call out those security impacts (or non-impacts) though to prompt reviewers to think about ways it might be abused.

Reply:
2e1c7200a4651bb31f3bd00446653d985e2a6ba1